Gordon360/AuthorizationFilters/StateYourBusiness.cs

```diff
@@ -192,6 +193,18 @@ private bool canReadOne(string resource)
 
                         return false;
                     }
+                case Resource.HOUSING:
+                    {
+                        // The members of the apartment application can only read their application
+                        var housingService = new HousingService(new UnitOfWork());
+                        var sess_cde = Helpers.GetCurrentSession().ToString();
+                        int? applicationID = housingService.GetApplicationID(user_id, sess_cde);
```

Comment:
I think this might cause problems if the housing admin tries to get data, and it will also have wrong behavior in the case were the current user is not on any applications (in there case, the desired behavior is a 404 not found, but I think this gets checked first, meaning you would get an "unauthorized" code instead of a "not found" error code)
I would suggest removing this from the the `GetApartmentApplication`, or have it return false only if the result of `Get ApplicationID` is not null and is not equal to the applicationID that was passed (I don't know if it is possible for the StateYourBusiness to access the parameter of the endpoint in the controller class, so this second idea may not be possible)

Reply:
So would it return true otherwise? I can set it to return false only if `(applicationID.HasValue && applicationID != requestedApplicationID)`